src/main/java/in/projecteka/gateway/common/CentralRegistry.java

```diff
@@ -0,0 +1,20 @@
+package in.projecteka.gateway.common;
+
+import in.projecteka.gateway.clients.ClientRegistryClient;
+import in.projecteka.gateway.clients.ClientRegistryProperties;
+import lombok.AllArgsConstructor;
+import reactor.core.publisher.Mono;
+
+import static java.lang.String.format;
+
+@AllArgsConstructor
+public class CentralRegistry {
+    private final ClientRegistryClient clientRegistryClient;
+    private final ClientRegistryProperties properties;
+
+    public Mono<String> authenticate() {
+        return clientRegistryClient
+                .getTokenFor(properties.getClientId(), properties.getXAuthToken())
+                .map(session -> format("%s %s", session.getTokenType(), session.getAccessToken()));
+    }
```

Comment:
Separate PR needs to be raised for caching the token, so that we will not make a call to CR always.